src/pt/huntersscans/src/eu/kanade/tachiyomi/extension/pt/huntersscans/HuntersScans.kt

```diff
@@ -1,23 +1,247 @@
+        return try {
+            val mangaPaged = url.newBuilder()
+                .addQueryParameter("page", "$page")
+                .build()
+            chapterFromJScript(client.newCall(GET(mangaPaged, headers)).execute())
+        } catch (e: Exception) {
+            Log.e("HuntersScans", e.toString())
+            emptyList()
+        }
+    }
+
+    private fun chapterFromJScript(response: Response): MutableList<SChapter> {
+        return try {
+            val jScript = response.asJsoup().select(chapterListSelector())
+                .map { element -> element.html() }
+                .filter { element -> element.isNotEmpty() }
+                .first { chapterRegex.find(it) != null }
+
+            val chaptersLinks = chapterRegex.findAll(jScript)
+                .map { result -> result.groups.map { it?.value } }
+                .flatMap { it }
+                .toSet()
+
+            chaptersLinks.map { chapterLink ->
+                SChapter.create().apply {
+                    name = chapterLink!!.toChapterName()
+                    setUrlWithoutDomain(chapterLink.toChapterAbsUrl())
+                }
+            }.toMutableList()
+        } catch (e: Exception) {
+            Log.e("HuntersScans", e.toString())
+            mutableListOf()
+        }
+    }
+
+    private fun containsDuplicate(chapters: List<SChapter>): Boolean {
+        return chapters.size != chapters.distinctBy { it.name }.size
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val chapters = chapterFromJScript(response)
+        val alwaysVisibleChapters = mutableSetOf<SChapter>()
+
+        val origin = response.request.url
+        var nextPage = 2
+
+        do {
+            if (chapters.size < 2) { break }
+
+            chapters += fetchChapterRequest(origin, nextPage).sortedBy { it.name.toInt() }
+
+            alwaysVisibleChapters += chapters.removeFirst()
+            alwaysVisibleChapters += chapters.removeLast()
+
+            nextPage++
+        }
+        while (!containsDuplicate(chapters) && chapters.isNotEmpty())
+
+        chapters += alwaysVisibleChapters
+
+        return chapters
+            .distinctBy { it.name }
+            .sortedBy { it.name.toInt() }
+            .reversed()
+    }
+
+    override fun chapterListSelector() = "script"
+
+    override fun imageUrlParse(document: Document) = ""
+
+    override fun latestUpdatesFromElement(element: Element) =
+        SManga.create().apply {
+            title = element.selectFirst("h3")!!.ownText()
+            thumbnail_url = element.selectFirst("img")?.absUrl("src")
+            setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        }
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> =
+        super.fetchLatestUpdates(page).map { it.removeNovels() }
```

Comment:
move this to `latestUpdatesParse`

Reply:
Resolved: https://github.com/keiyoushi/extensions-source/pull/1650/commits/2d13eba4008bb137a9ceec5742808dbd3b708a13